dashboard/app/controllers/reports_controller.rb

```diff
@@ -75,17 +75,19 @@ def search_for_teachers
     end
 
     # Prune the set of fields to those that will be displayed.
+    @teacher_limit = 100
     @headers = ['ID', 'Name', 'Email', 'Address', 'Num Students']
-    @teachers = @teachers.pluck('id', 'name', 'email', 'full_address', 'COUNT(followers.id) AS num_students')
+    @teachers = @teachers.limit(@teacher_limit).pluck('id', 'name', 'email', 'full_address', 'COUNT(followers.id) AS num_students')
```

Comment:
If you are able to perform those joins through activerecord, I believe it is clever enough to evaluate the queries very lazily